libraries/chain/block_header_state.cpp

```diff
@@ -61,13 +62,29 @@ block_header_state block_header_state::next(block_header_state_input& input) con
    result.header = block_header {
       .timestamp         = input.timestamp, // [greg todo] do we have to do the slot++ stuff from the legacy version?
       .producer          = input.producer,
+      .confirmed         = hs_block_confirmed, // todo: consider 0 instead
       .previous          = input.parent_id,
       .transaction_mroot = input.transaction_mroot,
       .action_mroot      = input.action_mroot,
-      //.schedule_version = ?, [greg todo]
-      //.new_producers = ?     [greg todo]
+      .schedule_version  = header.schedule_version
    };
 
+   if(!proposer_policies.empty()) {
+      auto it = proposer_policies.begin();
```

Comment:
Do we want the earliest one?

Reply:
They are ordered by timestamp. We want the earliest timestamp for our comparison with current (input.timestamp).